src/sql/src/plan/statement/ddl.rs

```diff
@@ -3761,27 +3766,33 @@ pub fn plan_alter_source(
     match action {
         AlterSourceAction::SetOptions(options) => {
             let CreateSourceOptionExtracted {
-                size: size_opt,
+                seen: _,
                 remote: remote_opt,
-                ..
+                size: size_opt,
+                timeline: timeline_opt,
             } = CreateSourceOptionExtracted::try_from(options)?;
 
+            if let Some(value) = remote_opt {
+                remote = AlterSourceItem::Set(value);
+            }
             if let Some(value) = size_opt {
                 size = AlterSourceItem::Set(value);
             }
-
-            if let Some(value) = remote_opt {
-                remote = AlterSourceItem::Set(value);
+            if let Some(_) = timeline_opt {
+                sql_bail!("Cannot modify the TIMELINE of a SOURCE.");
```

Comment:
unfortunate that we have to do this here, perhaps we can add a `TODO: split CreateSourceOptionExtracted into alterable and non-alterable structs`

Reply:
_shrug_... sort of unfortunate, but i'm hesitant to generalize from one example! (I think that may require changes to the macro, and it makes it harder to validate whether all options are consumed.)

Reply:
there will be at least 2 more examples of this, `epoch_timestamp_ms`, and `TIMESTAMP GRANULARITY`

Reply:
we can add the todo when we add those